Practice Project/Assets/EnemyEagle.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyEagle : Enemy
+{
+    public Transform topPoint;
+    public Transform bottomPoint;
+    public float moveSpeed = 5;
+
+    private Rigidbody2D rb;
+    private Collider2D col;
+    private bool isUp = true;
+    private float topY;
+    private float bottomY;
+    private System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
+
+    protected override void Start()
+    {
+        base.Start();
+        rb = GetComponent<Rigidbody2D>();
+        col = GetComponent<Collider2D>();
+
+        transform.DetachChildren();
+        topY = topPoint.position.y;
+        bottomY = bottomPoint.position.y;
+    }
+
+    void Update()
+    {
+       if (isUp)
+        {
+            rb.velocity = new Vector2(rb.velocity.x, moveSpeed);
+                
+            if (transform.position.y > topY)
+            {
+                isUp = false;
+            }
+        }
+        else if (!isUp)
+        {
+            rb.velocity = new Vector2(rb.velocity.x, -moveSpeed);
+            
+            if (transform.position.y < bottomY)
+            {
+                isUp = true;
+            }
+        }
+        
+        if(sw.ElapsedMilliseconds > (long)600)
```

Comment:
Stopwatch is a _little_ overkill here. It doesn't really matter much here because we're talking about just a couple enemies, but this same thing can be done with just a time tracking variable, like a float or something. No need to change it, just food for thought.